tsdb/agent/series.go

```diff
@@ -45,43 +45,73 @@ func (m *memSeries) updateTimestamp(newTs int64) bool {
 	return false
 }
 
-// seriesHashmap is a simple hashmap for memSeries by their label set.
-// It is built on top of a regular hashmap and holds a slice of series to
-// resolve hash collisions. Its methods require the hash to be submitted
+// seriesHashmap lets agent find a memSeries by its label set, via a 64-bit hash.
+// There is one map for the common case where the hash value is unique, and a
+// second map for the case that two series have the same hash value.
+// Each series is in only one of the maps. Its methods require the hash to be submitted
 // with the label set to avoid re-computing hash throughout the code.
-type seriesHashmap map[uint64][]*memSeries
+type seriesHashmap struct {
```

Comment:
Let's start to be strict in PRs given we are close to merging our branch to production code. This looks like unrelated change, thus should be in separate PR to me 🤔

Reply:
Oh I think it's because you merged main accidentally? Let's split that perhaps?

Reply:
eh maybe whatever, I will merge all using non squash button, will be same result

Reply:
🙈